### DISCUSSION TITLE
What is the name of the feature that interpolates characters?

### DISCUSSION THREAD

**Discussion by ClaireCJS** [microsoft/terminal#17956] [General]
For example, if you do "~~", it makes the 2nd tilde build off the first.

If you do "***", it makes the 2nd asterisk higher [the only one i hate haha]

If you do "-->", it re-draws the "--" to look like "—"  so you get "—>" {but it fully connects the horizontal line to the arrowhead}

There's a bunch of them. I kind of had to discover them one by one. It's a big part of Windows Terminal that I really liked.

Just.. What is that feature called?

Because somehow it stopped, but i don't know what it's called to turn it back on

**Comment by lhecker** [comment]
They're called ligatures: https://en.wikipedia.org/wiki/Ligature_(writing)

Originally, they were used to make text more similar to handwriting, but they're crucial for more complex scripts (like Indic languages/scripts). I'm not entirely sure what Monospace font started it (maybe Fira Code?), but someone got the idea at some point that you could use ligatures to turn common code sequences like `!=` into `≠` and so on.

Windows Terminal ships with Cascadia Code which has these ligatures: https://github.com/microsoft/cascadia-code#font-features
If it stopped working for you, you may be using a different font now (Cascadia Mono for instance), or you disabled it with the font feature settings.

**Reply by ClaireCJS** [reply]
> They're called ligatures: 

Thanks!  I should have realized that.  

Really nice to see the official list though! Bookmarking!


>If it stopped working for you, you may be using a different font now (Cascadia Mono for instance), or you disabled it with the font feature settings.


Ahh that was it! I must have changed from Cascadia Code to Cascadia Mono at some point, and not noticed. Thanks!

**Comment by ClaireCJS** [comment]
Hey, any clue how to enable the cursive ligatures (```                    "editor.fontLigatures": "'calt', 'ss01'"```) in Windows Termianl's JSON file? I can't quite figure that out.

**Reply by lhecker** [reply]
Like so:
```json
{
    "profiles":
    {
        "<profile name or 'defaults'>":
        {
            "font":
            {
                "face": "Cascadia Code",
                "features":
                {
                    "ss01": 0
                }
            }
        }
    }
}
```

There's a settings UI for it nowadays:

![image](https://github.com/user-attachments/assets/86c0a839-a9bb-4cff-a199-19e2dbc0d123)

...but I just noticed that we don't show ss01, because we don't check for font features exclusive to the italic variant. We should fix that.
